common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt

```diff
@@ -0,0 +1,1885 @@
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 50
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:friendly
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 80
+		}
+		modifier = {
+			trigger = {
+				imperia_frankfurt_parliament_law_is_preferred_by_dominating_left_ai_trigger = no
+				has_variable = frankfurt_parliament_voted_monarchy
+			}
+			multiply = 0
+		}
+		modifier = {
+			trigger = {
+				imperia_frankfurt_parliament_law_is_preferred_by_dominating_left_ai_trigger = yes
+			}
+			add = 20
+			multiply = 5
+		}
+	}
+	possible = {
+		custom_tooltip = {
+			text = frankfurt_parliament_voting_governance_principle
+			global_var:frankfurt_parliament_current_vote = 1
+		}
+		hidden_trigger = {
+			NOT = {
+				has_variable = frankfurt_parliament_voted_republic
+			}
+		}
+	}
+	selected = {
+		has_variable = frankfurt_parliament_voted_republic
+	}
+	effect = {
+		set_variable = frankfurt_parliament_voted_republic
+		## Remove this voting var (if it exists, thats what the ?= means)
+		remove_variable ?= frankfurt_parliament_voted_monarchy
+		custom_tooltip = {
+			text = frankfurt_votes_for_republic_tt
+		}
+	}
+}
+
+je_frankfurt_parliament_vote_monarchy_button = {
+	name = "je_frankfurt_parliament_vote_monarchy_button"
+	desc = "je_frankfurt_parliament_vote_monarchy_button_desc"
+	visible = {
+		has_global_variable = frankfurt_parliament_current_vote
+		global_var:frankfurt_parliament_current_vote = 1
+	}
+	ai_chance = {
+		base = 0
+		modifier = {
+			trigger = {
+				ruler = {
+					interest_group = {
+						AND = {
+							prefers_law = {
+								law = law_type:law_monarchy
+								comparison_law = law_type:law_presidential_republic
+							}
+							prefers_law = {
+								law = law_type:law_monarchy
+								comparison_law = law_type:law_parliamentary_republic
+							}
+						}
+					}
+				}
+			}
+			add = 75
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:cordial
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_monarchy
+				}
+			}
+			add = 20
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:amicable
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_monarchy
+				}
+			}
+			add = 50
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:friendly
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_monarchy
+				}
+			}
+			add = 80
+		}
```

Comment:
How do these even work? The conditions are cyclical... Is this only for players?